### PR TITLE
[Fix] Storybook HMR

### LINKED ISSUE #7460: 🐛 Storybook Hot Reload Broken
## 🐛 Bug

Storybook hot reload no longer works after upgrading to V7.  A change to the story results in a "Couldn't find story" error.  The only resolution is to full stop Storybook and restart it.

## ✅ Acceptance Criteria

- [ ] If Storybook is already running, when a change is made it should automatically perform a hot reload and show the change.

### PR DESCRIPTION
**Pull request by esizer**
🤖 Resolves #7460 

## 👋 Introduction

This does a couple of things to increase the stability of HMR in storybook.

- Updates the namespace (there is another package by the name `storybook-helpers`)
- Fixes the default export declaration for `main.ts` config file
- Replaces the custom h2 hooks with the `HydrogenWebpackPlugin`
- Bumps storybook to latest version

## 🕵️ Details

I'm not exactly sure the cause of the issue but it was somewhere in our custom hydrogen hooks. These hooks proved to be less than reliable when regenerating hydrogen causing specific stories to crash when changed. 

We have a more stable webpack plugin that I was able to replace it with while also making some slight tweaks to the storybook webpack config to match [the recommendations](https://storybook.js.org/docs/react/builders/webpack). 

## 🧪 Testing

> **Note**
> When making changes, try to make them hydrogen styles you know do not exist to ensure it is building properly and not just reading existing style rules.

1. Update depedencies `npm install`
2. Run storybook `npm run storybook`
3. Confirm hydrogen still builds on initial run
4. Make a change to component 
5. Confirm HMR works and changes reflected
6. Make a change to the `*.stories.tsx` file
7. Confirm HMR works and changes reflected

### CODE REVIEW

File: apps/web/.storybook/main.ts [hunk 1]
```diff
@@ -1,3 +1,3 @@
-import config from "storybook-helpers/main";
+import config from "@gc-digital-talent/storybook-helpers/main";
 
-export default config;
+export default { ...config };
```
Comment (esizer) [inline apps/web/.storybook/main.ts:3]: While this wasn't breaking anything, there was an annoying warning about re-exporting as default. Deconstruction avoids this warning.

**Comment by substrae** [comment]
Working for me!

**Review by vd1992 — APPROVED** [review]
I see Matt is also reviewing but I gave it a go too. 

Chromatic deploy successful. not sure why this registered a diff, missing decorator or something? Or a change picked up from a different merge?

image

Storybook builds locally, updating stuff in a story file causes a successful reload. 

Amazing!

**Comment by esizer** [comment]
> Chromatic deploy successful. not sure why this registered a diff, missing decorator or something? Or a change picked up from a different merge?

Yeah, I'm not sure what happened there because they are there locally when I build 😢 Looks like [the pending component needs to be stabilized](https://github.com/GCTC-NTGC/gc-digital-talent/issues/7742) as well.

**Comment by vd1992** [comment]
> > Chromatic deploy successful. not sure why this registered a diff, missing decorator or something? Or a change picked up from a different merge?
> 
> Yeah, I'm not sure what happened there because they are there locally when I build 😢 Looks like [the pending component needs to be stabilized](https://github.com/GCTC-NTGC/gc-digital-talent/issues/7742) as well.

Hm, given neither are new components seems like something somewhere knocked something loose. Maybe a dependency update slightly changed behaviour of something 🙃 
Oh well, just some noise to put up with for now

**Comment by esizer** [comment]
> Hm, given neither are new components seems like something somewhere knocked something loose. Maybe a dependency update slightly changed behaviour of something 🙃
> Oh well, just some noise to put up with for now

yeahh, I figured we'd take missing triangles in 2 stories over broken HMR 😅

**Review by mnigh — APPROVED** [review]
So good!

**Comment by substrae** [comment]
Thanks for the speedy merge folks ❤️

**Pull request merged**